curvefs/src/tools/curvefs_tool.h

```diff
@@ -102,16 +140,82 @@ class CurvefsToolRpc : public CurvefsTool {
         request_ = std::make_shared<RequestT>();
         response_ = std::make_shared<ResponseT>();
         service_stub_ = std::make_shared<ServiceT>(channel_.get());
+        AddUpdateFlagsFuncs();
+        UpdateFlagsFromConf();
         return 0;
     }
 
+    virtual void AddUpdateFlagsFunc(
+        const std::function<void(curve::common::Configuration*,
+                                 google::CommandLineFlagInfo*)>& func) {
+        updateFlagsFunc_.push_back(func);
+    }
+
+    virtual void UpdateFlagsFromConf() {
```

Comment:
it seems no need to be virtual for these two functions.

Reply:
> it seems no need to be virtual for these two functions.

UpdateFlagsFromConf sets as a virtual function to prevent some subclasses from needing to manually manage FlagsInfo by themselves. It is indeed not necessary.